### ISSUE TITLE
Generated `mod.rs` in `OUT_DIR` doesn't look in relative path for `mod` statements.

### ISSUE THREAD

**Issue by brettfo** [rust-analyzer/rust-analyzer#4075]
REPRO:
1.  Clone https://github.com/ixmilia/dxf-rs at commit 9a01d992f3e394d84131143973270cd0a5191d3d
2.  Open VS Code with rust-analyzer installed (v 0.2.143)
3.  Navigate to `src/drawing.rs`
4.  Hover cursor at line 74, character 24 (in `Entity` identifier).

RESULT:
No hover information is provided, and the identifier was not colored as a type.

EXPECTED:
Hover text should be provided, much like on line 71 for the identifier `Block`.

My best analysis is the failure is due to the `struct Entity` being defined in generated code produced by `build/build.rs`.  The type `Entity` is defined in file `target/debug/build/dxf-.../out/generated/entities.rs`, line 60.  That file is included in the build via `src/generated.rs` with contents:

``` rust
include!(concat!(env!("OUT_DIR"), "/generated/mod.rs"));
```

`.vscode/settings.json` contains the settings:

``` json
"rust-analyzer.cargo.loadOutDirsFromCheck": true,
"rust-analyzer.procMacro.enabled": true
```

Setting `RUST_LOG=info` results in this:

<details>
<summary>(82 lines)</summary>

```
[INFO  rust_analyzer::main_loop] loop turn = Request(Request { id: RequestId(U64(45)), method: "textDocument/hover", params: Object({"position": Object({"character": Number(24), "line": Number(73)}), "textDocument": Object({"uri": String("file:///e%3A/ixmilia/dxf-rs/src/drawing.rs")})}) })
[INFO  salsa::derived::slot] LineIndexQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] LineIndexQuery(FileId(99)): returning memoized value changed at R474
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): invoked at R4874
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): returning memoized value changed at R477
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): invoked at R4874
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): invoked at R4874
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): returning memoized value changed at R477
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): invoked at R4874
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): returning memoized value changed at R4870
[INFO  rust_analyzer::main_loop] loop turn = Response { id: RequestId(U64(45)), error: None }
[INFO  rust_analyzer::main_loop] handled req#45 in 413.7µs
[INFO  rust_analyzer::main_loop] loop turn = Request(Request { id: RequestId(U64(46)), method: "textDocument/hover", params: Object({"position": Object({"character": Number(24), "line": Number(73)}), "textDocument": Object({"uri": String("file:///e%3A/ixmilia/dxf-rs/src/drawing.rs")})}) })
[INFO  salsa::derived::slot] LineIndexQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] LineIndexQuery(FileId(99)): returning memoized value changed at R474
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): invoked at R4874
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): returning memoized value changed at R477
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): invoked at R4874
[INFO  salsa::derived::slot] ParseQuery(FileId(99)): returning memoized value changed at R969
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): invoked at R4874
[INFO  salsa::derived::slot] AstIdMapQuery(HirFileId(FileId(FileId(99)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): invoked at R4874
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): returning memoized value changed at R4870
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): invoked at R4874
[INFO  salsa::derived::slot] SourceRootCratesQuery(SourceRootId(95)): returning memoized value changed at R477
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): invoked at R4874
[INFO  salsa::derived::slot] CrateDefMapQueryQuery(CrateId(50)): returning memoized value changed at R4870
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): invoked at R4874
[INFO  salsa::derived::slot] GenericParamsQuery(AdtId(StructId(StructId(1556)))): returning memoized value changed at R4870
[INFO  rust_analyzer::main_loop] loop turn = Response { id: RequestId(U64(46)), error: None }
[INFO  rust_analyzer::main_loop] handled req#46 in 378.3µs
```
</details>

Edit: rls is able to parse `include!`d files.  Machine is Windows.

**Comment by edwin0cheng** [comment]
Okay, I think I found the problem : 

Your project generated only `include` `mod.rs` in `OUT_DIR` but also generated `header.rs` etc, and it depends on the code inside `mod.rs` to import them: 

```
// mod.rs
mod header;
...
//

However, currently rust-analyzer only support single file `include!`, i.e. : it do not search files relative to included file.

**Comment by edwin0cheng** [comment]
Okay, I think I found the problem : 

Your project only call `include` for `mod.rs` in `OUT_DIR`,  but also generated `header.rs` etc and  depends on the code inside `mod.rs` to import them: 

```rust
// mod.rs
mod header;
...
```

However, currently rust-analyzer only support single file `include!`, i.e. : it does not search files relative to included file.

**Comment by edwin0cheng** [comment]
closed this one in favor of  #7846.
